### ISSUE TITLE
SR-17-0000386 Citation data transfer to Police Records

### ISSUE THREAD

**Issue by RikiSjachrani-COM** [City-of-Madison/Development-Team-2#12]
Create an export of Accela Citations to be imported into the Police Record Management System

To reduce manual entry into the Police RMS, improve accuracy of data being transferred, save staff time to focus on other tasks

To accomplish this - Not sure our approach just yet, going to be a specification document, SSIS package or run-able report…

We will know we are making progress when we have specifications and a mock test file to import and know this is completed when Accela citations are smoothly being processed into the Police RMS without manual data entry

This should be completed by end of the quarter and we should be halfway done with with a test import and/or manual import into Police RMS

Resource wise, try to allocate - Hour a day

Next Actionable Steps:

- [x] Allocate time on Calendar
- [x] Meet with agency SME
- [ ] Determine fields for import
- [ ] Specifications for import
- [ ]  Determine mechanism to output xml - Chris
- - [ ] output file stuff
- - [ ] attachment embedded in output file - @KimGrittner-COM you might be a resource on this
- [ ] Test file
- [ ] …

Materials to reference 
\\Fps1\data1\Information Technology\Applications and Systems\Accela\Active Projects\SR-17-0000386 Citation data transfer to Police Records

**Comment by RikiSjachrani-COM** [comment]
@ChrisBilik-COM Sharon has asked that you participate on this Service Request, it's still in the dev spec phase but feel free to schedule a meeting sometime next week to discuss this project

**Comment by ChrisBilik-COM** [comment]
@RikiSjachrani-COM  Hey Riki can you find out the code they use to generate the sample xml files...looking for something along the lines of: FOR XML path ('Ticket'), ROOT ('Tickets'), ELEMENTS XSINIL.  I want to know what output format they are using so I can mimic it as close to possbile

**Comment by ChrisBilik-COM** [comment]
@RikiSjachrani-COM  Hey Riki can you find out the code they use to generate the sample xml files...looking for something along the lines of: FOR XML path ('Ticket'), ROOT ('Tickets'), ELEMENTS XSINIL.  I want to know what output format they are using so I can mimic it as close to possible.

**Comment by ChrisBilik-COM** [comment]
I have the fields that have been mapped set in a sql statement to return as xml.  Only gotcha is the Violator/TicketRecipient address....(B3Contact) the address field is free text.  so it would need to be mega parsed...spoke with the folks here about it and not pretty.  Maybe we can map the address just as one...city state and zip are still separate.
![sample1](https://user-images.githubusercontent.com/32718943/32910222-493756a6-cace-11e7-9738-69cab3b7d01f.jpg)

**Comment by RikiSjachrani-COM** [comment]
@ChrisBilik-COM so we can't import duplicate citation numbers right?

**Comment by ChrisBilik-COM** [comment]
@RikiSjachrani-COM correct...if I put an xml file with the same citation number, the interface will not import the data and gives an error to the extent of record already exists

**Comment by RikiSjachrani-COM** [comment]
@ChrisBilik-COM ok, I think I got the query in a decent place, it's not fool proof but just looking through the records I think it should work since the fields that we are passing on don't change very much between duplicate citations. I have it working from a Thursday 6:00 am to previous Thursday 6:00 am. I also feel like it would be good to either make this a view or stored procedure to make it easier to manage and run from the Report and SSIS package. 

We might want to discuss a bit in person.

**Comment by ChrisBilik-COM** [comment]
@RikiSjachrani-COM  lets try the view idea.  I need to see how SSIS will handle because it was using a temp table...I don't think it should matter so lets get that created and Ill work on verifying the xml output is still good.  From there we can do some decent front and back end testing....minus the address part.

**Comment by RikiSjachrani-COM** [comment]
Do you have the elam dev account password? If you give me the sql, I will create the view in Dev and Test (I'll point Test to the Replication Database)

**Comment by RikiSjachrani-COM** [comment]
ok, I created the view it's called vwWeeklyCitations

**Comment by ChrisBilik-COM** [comment]
All good on SSIS side using the view...33 xml files created for the date range specified.

**Comment by RikiSjachrani-COM** [comment]
So from my notes we still need to sort out the "Ticket Type" so that it's Building Inspection etc...and the court date time?

**Comment by ChrisBilik-COM** [comment]
Ticket Type yes, court time I did based off of Crystal logic
   CASE when
	    right(cast(vwVC.CNumber/2 AS decimal(8,2)),2)=50 
		THEN '09:30:00 AM'
		ELSE '08:30:00 AM'END AS MDTY_APRN_TM,

**Comment by RikiSjachrani-COM** [comment]
- [ ] Contact Time and Date are mixed up
 - [ ] HIDDENUSERFN and HIDDENUSERLN could be filled in
 - [ ] INCD_RPT_NARR could be mapped to the inspection descript
 - [ ] 
 - [ ] 
 - [ ]

**Comment by ChrisBilik-COM** [comment]
inspection decript....is that the same as description of violation?  other 2 are done

**Comment by RikiSjachrani-COM** [comment]
Put this in the Ice Box until there's some movement on the import issues with Tyler

**Comment by RikiSjachrani-COM** [comment]
@JimSchmitt-COM there's quite a bit here to start with...the basic premise is Accela can do Citations but these have to be manually entered into the Police RMS (Record Management system), this project is to create an export to a monitored folder that LERMS aka the PD RMS, can import. 

Once this is established, we can expand to other agencies that are looking to do citation out of Accela. That is a related SR that I will also be assigning to you.
